### PR TITLE
[v2][adjuster] Implement adjuster for sorting attributes and events

### PR DESCRIPTION
**Pull request by mahadzaryab1**
## Which problem is this PR solving?
- Towards #6344

## Description of the changes
- Implemented an adjuster to sort attributes and events that operates natively on the OTLP data format. This is analogous to the `SortTagsAndLogFields` adjuster in v1.

## How was this change tested?
- Added a unit test

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `npm run lint` and `npm run test`

### CODE REVIEW

File: cmd/query/app/querysvc/adjuster/sort.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		resource := rs.Resource()
+		s.sortAttributes(resource.Attributes())
+		scopeSpans := rs.ScopeSpans()
+		for j := 0; j < scopeSpans.Len(); j++ {
+			ss := scopeSpans.At(j)
+			spans := ss.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				s.sortAttributes(span.Attributes())
+				s.sortEvents(span.Events())
```
Comment (mahadzaryab1) [inline cmd/query/app/querysvc/adjuster/sort.go:39]: @yurishkuro Wanted a sanity check here. In v1, the "event" field was moved to the first position (ref: https://github.com/jaegertracing/jaeger/blob/main/model/adjuster/sort_tags_and_log_fields.go#L31). However, in v2, logs are stored as event attributes and the "event" attribute is stored as the event name (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/translator/jaeger/jaegerproto_to_traces.go#L406).

Comment (yurishkuro) [reply cmd/query/app/querysvc/adjuster/sort.go:39]: sgtm. Good to have first class fields.

File: cmd/query/app/querysvc/adjuster/sort.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+func (s SortAttributesAndEventsAdjuster) Adjust(traces ptrace.Traces) {
+	resourceSpans := traces.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		rs := resourceSpans.At(i)
+		resource := rs.Resource()
+		s.sortAttributes(resource.Attributes())
+		scopeSpans := rs.ScopeSpans()
+		for j := 0; j < scopeSpans.Len(); j++ {
+			ss := scopeSpans.At(j)
```
Comment (yurishkuro) [inline cmd/query/app/querysvc/adjuster/sort.go:34]: why not sort scope attributes?
```suggestion
			ss := scopeSpans.At(j)
			s.sortAttributes(ss.Attributes())
```

**Review by yurishkuro — APPROVED** [review]

File: cmd/query/app/querysvc/adjuster/sort.go (hunk 2 above)
Comment (mahadzaryab1) [reply cmd/query/app/querysvc/adjuster/sort.go:34]: @yurishkuro this is something I wanted to ask. Do we want to sort all possible attributes? or just the ones that are used in model to otel translation?

**Pull request merged**